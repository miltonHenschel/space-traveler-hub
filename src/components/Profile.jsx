import { useSelector } from 'react-redux';
import styles from '../styles/Profile.module.css';

function Profile() {
  const missions = useSelector((state) => state.missions);

  const reservedMissions = missions.missionsItems.filter(
    (mission) => mission.reserved,
  );

  return (
    <section className={styles.section}>
      <div className={styles.div}>
        <table className={styles.table}>
          <thead>
            <tr>
              <th className={styles.th}>My Rockets</th>
            </tr>
          </thead>
          <tbody>
            {reservedMissions.length > 0 ? (
              reservedMissions.map((mission) => (
                <tr key={mission.mission_id}>
                  <td className={styles.td}>{mission.mission_name}</td>
                </tr>
              ))
            ) : (
              <tr>
                <td className={styles.td}>
                  You don&apos;t have reserved missions yet.
                </td>
              </tr>
            )}
          </tbody>
        </table>
      </div>
      <div className={styles.div}>
        <table className={styles.table}>
          <thead>
            <tr>
              <th className={styles.th}>My Missions</th>
            </tr>
          </thead>
          <tbody>
            {reservedMissions.length > 0 ? (
              reservedMissions.map((mission) => (
                <tr key={mission.mission_id}>
                  <td className={styles.td}>{mission.mission_name}</td>
                </tr>
              ))
            ) : (
              <tr>
                <td className={styles.td}>
                  You don&apos;t have reserved missions yet.
                </td>
              </tr>
            )}
          </tbody>
        </table>
      </div>
    </section>
  );
}

export default Profile;
