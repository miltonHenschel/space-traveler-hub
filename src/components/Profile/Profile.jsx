import React from 'react';
import { useSelector } from 'react-redux';
import styles from '../../styles/Profile.module.css';

function Profile() {
  const missions = useSelector((state) => state.missions);
  const reservedMissions = missions.missionsItems.filter(
    (mission) => mission.reserved,
  );

  const Rockets = useSelector((state) => state.rockets);
  const ReservedRockets = Rockets.filter((rocket) => rocket.reserved === true);

  return (
    <section className={styles.section}>
      <div className={styles.div}>
        <table className={styles.table}>
          <thead>
            <tr>
              <th className={styles.th}>My Missions</th>
            </tr>
          </thead>
          <tbody>
            {reservedMissions.length > 0 ? (
              reservedMissions.map((mission) => (
                <tr key={mission.mission_id}>
                  <td className={styles.td}>{mission.mission_name}</td>
                </tr>
              ))
            ) : (
              <tr>
                <td className={styles.td}>
                  You haven&apos;t joined any missions yet.
                </td>
              </tr>
            )}
          </tbody>
        </table>
      </div>
      <div className={styles.div}>
        <table className={styles.table}>
          <thead>
            <tr>
              <th className={styles.th}>My Rockets</th>
            </tr>
          </thead>
          <tbody>
            {ReservedRockets.length > 0 ? (
              ReservedRockets.map((rocket) => (
                <tr key={rocket.id}>
                  <td className={styles.td}>{rocket.name}</td>
                </tr>
              ))
            ) : (
              <tr>
                <td className={styles.td}>
                  You don&apos;t have reserved rockets yet.
                </td>
              </tr>
            )}
          </tbody>
        </table>
      </div>
    </section>
  );
}

export default Profile;
