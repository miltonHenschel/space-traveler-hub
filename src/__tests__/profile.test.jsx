import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../components/Profile/Profile';
import store from '../redux/Store';

it('Check MyProfile renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
