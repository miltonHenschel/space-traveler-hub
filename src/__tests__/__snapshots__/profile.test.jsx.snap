// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check MyProfile renders correctly 1`] = `
<section
  className="section"
>
  <div
    className="div"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            className="th"
          >
            My Missions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="td"
          >
            You haven't joined any missions yet.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="div"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            className="th"
          >
            My Rockets
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="td"
          >
            You don't have reserved rockets yet.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</section>
`;
